pkg/generate/app/sourcelookup.go

```diff
@@ -90,6 +90,7 @@ type SourceRepository struct {
 	buildWithDocker  bool
 	ignoreRepository bool
 	binary           bool
+	dockerfile       bool
```

Comment:
From this struct's perspective, it doesn't know about the CLI. It's more about "was AddDockerfile called?"
Will think about how to make it better once I figure out the regression test.